inc/class-saml.php

```diff
@@ -443,12 +449,12 @@ public function samlAssertionConsumerService() {
 	 */
 	public function parseAttributeStatement() {
 		// Attributes
-		$attributes = $this->auth->getAttributesWithFriendlyName();
-		if ( ! isset( $attributes['uid'] ) ) {
-			$attributes = $this->auth->getAttributes();
-			if ( ! isset( $attributes['uid'] ) ) {
-				throw new \Exception( __( 'Missing SAML attributes: uid, mail', 'pressbooks-saml-sso' ) );
-			}
+		$attributes = $this->auth->getAttributes();
+		if (
+			! isset( $attributes[ self::SAML_MAP_FIELDS['uid'] ] ) ||
+			! isset( $attributes[ self::SAML_MAP_FIELDS['mail'] ] )
```

Comment:
@richard015ar I have a question about this extra conditional. Are plugin is meant to work if the client sends us a UID but chooses not to send a mail value, because we simply set the email to net_id@127.0.0.1 if no email value is received. Will changing this isset condition to require BOTh uid and mail change that?

Reply:
yes, no I think I added this additional check by mistake. I misunderstood the Exception message with the current business rule. I will change it, but I think before that, would be nice a quick review meeting to be sure it could works. Thank you for catching it, it defintely will break for those cases where we just get one of both attributes.

Reply:
@SteelWagstaff mail validation was removed